iocore/net/QUICNetVConnection.cc

```diff
@@ -451,14 +379,12 @@ QUICNetVConnection::start()
   this->_frame_dispatcher = new QUICFrameDispatcher(this);
 
   // Create frame handlers
-  QUICCCConfigQCP cc_config(this->_quic_config);
-  QUICLDConfigQCP ld_config(this->_quic_config);
   this->_pinger = new QUICPinger();
   this->_padder = new QUICPadder(this->netvc_context);
-  this->_rtt_measure.init(ld_config);
-  this->_congestion_controller = new QUICCongestionController(this->_rtt_measure, this, cc_config);
-  this->_loss_detector         = new QUICLossDetector(this, this->_congestion_controller, this->_pp_key_info, &this->_rtt_measure,
-                                              this->_pinger, this->_padder, ld_config, this->netvc_context);
+  this->_rtt_measure.init(this->_context->ld_config());
+  this->_congestion_controller           = new QUICCongestionController(*_context);
+  this->_context->_congestion_controller = this->_congestion_controller; // hijack cc into context
```

Comment:
Because this cc need to be set implicitly. It should be more abstraction like QUICCreditProvider or something else to provide credit. So we can avoid the bad `friend class`.

Reply:
I'm not sure how abstraction helps here.

Can we remove the `friend class` if we introduce abstract `QUICCongestionController` and `QUICCongestionControllerNewReno`?

LD needs CongestionController (it calls functions such as `on_packet_sent `, `process_ecn`), so just providing credits doesn't seem to satisfy the needs.

Reply:
Removed.